File: MlbDraft.API/Models/LeagueModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace MLBDraft.API.Models
{
    public class LeagueModel: LeagueAbstractModel
    {
        public Guid Id { get; set; }

        public override string Name { get; set; }

        public override int MinTeams { get; set; }

        public override int MaxTeams { get; set; }

        public IList<TeamModel> Teams { get; set; } = new List<TeamModel>();

        public IList<DraftModel> Drafts {get; set; } = new List<DraftModel>();

            
    }
}

File: MlbDraft.Client/MlbDraft/src/app/shared/models/league.model.ts
import { Team } from './team.model';
import { Draft } from './draft.model';

import {Guid } from 'guid-typescript';

export class League {
    public id: Guid;
    public name: string;
    public minTeams: number;
    public maxTeams: number;
    public teams : Team[];
    public drafts: Draft[];


    constructor(id: Guid, name: string, minTeams: number, maxTeams: number, teams: Team[], drafts: Draft[]) {
        this.id = id;
        this.name = name;
        this.minTeams = minTeams;
        this.maxTeams = maxTeams;
        this.teams = teams;
        this.drafts = drafts;
    
      }
}